Foundation/NSURLProtocol.swift

```diff
@@ -323,19 +439,65 @@ open class URLProtocol : NSObject {
         The only way that failure can occur is if the given class is not a
         subclass of NSURLProtocol.
     */
-    open class func registerClass(_ protocolClass: AnyClass) -> Bool { NSUnimplemented() }
-    
+    open class func registerClass(_ protocolClass: AnyClass) -> Bool {
+        if protocolClass is URLProtocol.Type {
+            _classesLock.lock()
+            guard !_registeredProtocolClasses.contains(where: { $0 === protocolClass }) else {
+                _classesLock.unlock()
```

Comment:
I rememberwhat @parkera had to say about this :)
https://github.com/apple/swift-corelibs-foundation/pull/902#issuecomment-283996792